API_Handlers/API_Handler.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Newtonsoft.Json.Linq;
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace Fabric_Metadata_Scanning
+{
+    public abstract class API_Handler
+    {
+        public string apiName { get; set; }
+        public UriBuilder apiUriBuilder { get; set; }
+        public JObject parameters { get; set; }
+
+        public API_Handler(string apiName)
+        {
+            this.apiName = apiName;
+            parameters = new JObject();
+            apiUriBuilder = new UriBuilder($"https://api.powerbi.com/v1.0/myorg/admin/workspaces/{apiName}");
+        }
+        
+        public abstract Task<object> run(string? scanId);
+
+        public async Task<HttpResponseMessage> sendGetRequest(string? scanId)
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                // Set the authorization header with the token
+                string accessToken = Auth_Handler.Instance.accessToken;
+                httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", accessToken);
+
+                // Detect usage - DO NOT MODIFY
```

Comment:
We can explain here that this is used for telemetry purposes to identify this specific client